bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemOffsetProfile.java

```diff
@@ -0,0 +1,137 @@
+    private static final String OFFSET_PARAM = "offset";
+
+    private final ProfileCallback callback;
+    private final ProfileContext context;
+
+    @Nullable
+    private QuantityType<?> offset;
+
+    public SystemOffsetProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.context = context;
+
+        Object paramValue = this.context.getConfiguration().get(OFFSET_PARAM);
+        logger.debug("Parameter '{}' has value '{}' ", OFFSET_PARAM, paramValue);
+        if (paramValue instanceof String) {
+            try {
+                offset = new QuantityType<>((String) paramValue);
+            } catch (IllegalArgumentException e) {
+                logger.error("Cannot convert value '{}' of parameter '{}' into a valid offset of type QyantityType.",
+                        paramValue, OFFSET_PARAM);
+            }
+        } else {
+            logger.error("Parameter '{}' is not of type String", OFFSET_PARAM);
+        }
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return SystemProfiles.OFFSET;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        logger.debug("State update from item: '{}'", state);
+        callback.handleCommand((Command) applyOffset(state, false));
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        logger.debug("Command from item: '{}'", command);
+        callback.handleCommand((Command) applyOffset(command, false));
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        logger.debug("Command from handler: '{}'", command);
+        callback.sendCommand((Command) applyOffset(command, true));
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        logger.debug("State update from handler: '{}'", state);
+        callback.sendUpdate((State) applyOffset(state, true));
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private Type applyOffset(Type state, boolean towardsThing) {
+        QuantityType finalOffset = offset;
+        if (finalOffset == null) {
+            logger.error("Offset not configured correctly, returning undef state.");
+            return UnDefType.UNDEF;
+        }
+
+        if (!towardsThing) {
+            finalOffset = finalOffset.negate();
+        }
+
+        Type result = UnDefType.UNDEF;
+
+        if (state instanceof QuantityType) {
+            QuantityType qtState = (QuantityType) state;
+            try {
+                if (finalOffset.getUnit() == AbstractUnit.ONE) {
+                    // allow offsets without unit -> implicitly assume its the same as the one from the state
+                    // TODO: do we want this? User would have to look up the original unit on the channel in the
+                    // binding...
+                    finalOffset = new QuantityType<>(finalOffset.toBigDecimal(), qtState.getUnit());
```

Comment:
This _can_ lead to quite funny results whenever the concrete units change. Or did I miss anything?

Reply:
That is why the TODO is in the code and in my comment on this PR ;-)